doc/dev/error-message-pitfalls-proposal.md

```diff
@@ -0,0 +1,76 @@
+# Introduction
+
+After the huge effort to harmonize and standardize our error message & code system, there is a great likelihood
+that any newly added message will again violate these standards. The only remedy as of now are good PR reviews
+which again are prone to human errors. I want to present common pitfalls I have seen when correcting
+these messages and afterwards show potential remedies which are open for discussion.
+
+# Problems
+
+1. Inconsistent Error Messages
+
+   1. Too advanced vocabulary
+
+      Many error and warning messages included words which are too advanced for normal users.
+      Examples are stat, normalize, glob, uncommon acronyms, etc.
+
+   2. Forgetting single quotes around %s expressions
+
+      As written in the error message design guideline, string formatters with %s have to be surrounded by single quotes so users
+      know where they start and end as variables with spaces do not make it clear otherwise.
+
+   3. Inconsistent sentence pattern
+
+      The error message design guideline states that sentences have to start with an upper case letter and should not contain a dot at the end.
+      Also exclamation marks should not be used as it is interpreted as shouting to the user.
+
+   4. Leaking internals
+
+      Some messages I have seen state which internal method has failed. E.g., in LCDproc there was a message `xosd: xosd_set_font() failed`. This though should be of no relevance for the user.
```

Comment:
Can you give a checklist for what is internal here.

Reply:
variable names and method names

Reply:
(please update in the document)